### ISSUE TITLE
MueLu: Cuda+UVM compilation failure

### ISSUE THREAD

**Issue by ndellingwood** [trilinos/Trilinos#12721]
## Bug Report
@trilinos/muelu

### Description
Attempting to compile MueLu in a Cuda build with UVM fails, it seems in code related to the `UncoupledAggregationFactory` - maybe an ETI-related issues? The error triggers a static_assert in Kokkos: "Incompatible View copy assignment"

```
[ 89%] Building CXX object packages/muelu/src/CMakeFiles/muelu.dir/Utils/ExplicitInstantiation/ETI_MueLu_UserAggregationFactory.cpp.o
/home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_opt-uvm/Trilinos/kokkos/core/src/Kokkos_View.hpp(1368): error: static assertion failed with "Incompatible View copy assignment"
          detected during:
            instantiation of "std::enable_if_t<Kokkos::Impl::ViewMapping<Kokkos::View<DataType, Properties...>::traits, Kokkos::View<RT, RP...>::traits, Kokkos::ViewTraits<DataType, Properties...>::specialize>::is_assignable_data_type, Kokkos::View<DataType, Properties...>> &Kokkos::View<DataType, Properties...>::operator=(const Kokkos::View<RT, RP...> &) [with DataType=unsigned int *, Properties=<Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>>, RT=uint32_t *, RP=<std::conditional_t<true, Kokkos::Cuda::array_layout, Kokkos::Cuda::array_layout>, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::Cuda::memory_space>, std::conditional_t<false, void, Kokkos::Experimental::DefaultViewHooks>>]" 
/home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_opt-uvm/Trilinos/packages/muelu/src/Graph/UncoupledAggregation/MueLu_UncoupledAggregationFactory_def.hpp(296): here
            instantiation of "void MueLu::UncoupledAggregationFactory<LocalOrdinal, GlobalOrdinal, Node>::Build(MueLu::UncoupledAggregationFactory<LocalOrdinal, GlobalOrdinal, Node>::Level &) const [with LocalOrdinal=int, GlobalOrdinal=longlong, Node=Tpetra_KokkosCompat_KokkosCudaWrapperNode]" 
/home/jenkins/weaver/workspace/KokkosEco_Trilinos_Weaver_CUDA112_opt-uvm/Trilinos/packages/muelu/src/Utils/MueLu_ETI_3arg.hpp(34): here
```

### Steps to Reproduce
1. SHA1: bdfdbc5420bd09458a0d67473863f22431ffbf03 
2. Configure: (Weaver rhel8 queue)
```bash
source /projects/ppc64le-pwr9-rhel8/legacy-env.sh
module purge
module load git cmake/3.23.1 cuda/11.2.2/gcc/8.3.1 openmpi/4.1.1/gcc/8.3.1/cuda/11.2.2 openblas/0.3.18/gcc/8.3.1
module load metis/5.1.0/gcc/8.3.1 hdf5/1.10.7/gcc/8.3.1/openmpi/4.1.1 parmetis/4.0.3/gcc/8.3.1/openmpi/4.1.1 zlib/1.2.11/gcc/8.3.1 boost/1.70.0/gcc/8.3.1
module list
export OMPI_CXX=$KOKKOS_DIR/bin/nvcc_wrapper

cmake \
 -D CMAKE_INSTALL_PREFIX=${TRILINOS_INSTALL_DIR} \
 -D CMAKE_CXX_COMPILER="`which mpicxx`" \
 -D CMAKE_CXX_FLAGS="-g" \
 -D CMAKE_C_COMPILER="`which mpicc`" \
 -D CMAKE_C_FLAGS="-g" \
 -D CMAKE_Fortran_COMPILER="`which mpifort`" \
 -D CMAKE_Fortran_FLAGS="-g" \
 -D CMAKE_BUILD_TYPE:STRING=RELEASE \
 -D CMAKE_CXX_STANDARD=17 \
 -D TPL_ENABLE_CUDA:STRING=ON \
 -D TPL_ENABLE_MPI:STRING=ON \
 -D MPI_BASE_DIR:PATH="$OPENMPI_ROOT" \
 -D MPI_BIN_DIR:PATH="$OPENMPI_BIN" \
 -D MPI_EXEC_POST_NUMPROCS_FLAGS:STRING="-map-by;socket:PE=4" \
 -D TPL_ENABLE_BLAS:STRING=ON \
  -D BLAS_LIBRARY_DIRS:FILEPATH="$OPENBLAS_ROOT/lib" \
  -D BLAS_LIBRARY_NAMES:STRING="openblas" \
 -D TPL_ENABLE_LAPACK:STRING=ON \
  -D LAPACK_INCLUDE_DIRS:FILEPATH="$OPENBLAS_ROOT/include" \
  -D LAPACK_LIBRARY_DIRS:FILEPATH="$OPENBLAS_ROOT/lib" \
  -D LAPACK_LIBRARY_NAMES:STRING="openblas" \
 -D TPL_ENABLE_Boost:BOOL=ON \
  -D Boost_INCLUDE_DIRS:FILEPATH="$BOOST_ROOT/include" \
  -D Boost_LIBRARY_DIRS:FILEPATH="$BOOST_ROOT/lib" \
 -D TPL_ENABLE_BoostLib:BOOL=ON \
  -D BoostLib_INCLUDE_DIRS:FILEPATH="$BOOST_ROOT/include" \
  -D BoostLib_LIBRARY_DIRS:FILEPATH="$BOOST_ROOT/lib" \
 -D TPL_ENABLE_Netcdf:BOOL=OFF \
 -D Trilinos_ENABLE_TESTS=OFF \
 -D Trilinos_ENABLE_EXAMPLES=OFF \
 -D Trilinos_ENABLE_Kokkos=ON \
 -D Kokkos_ENABLE_TESTS=ON \
  -D Kokkos_ENABLE_CUDA=ON \
  -D Kokkos_ENABLE_CUDA_LAMBDA=ON \
  -D Kokkos_ENABLE_CUDA_UVM=ON \
  -D Kokkos_ARCH_VOLTA70=ON \
  -D Kokkos_ARCH_POWER9=ON \
 -D Trilinos_ENABLE_KokkosKernels=ON \
 -D KokkosKernels_ENABLE_TESTS=ON \
  -D KokkosKernels_INST_MEMSPACE_CUDAUVMSPACE=ON \
  -D KokkosKernels_ENABLE_TPL_CUSOLVER=OFF \
 -D Trilinos_ENABLE_Tpetra=ON \
 -D Tpetra_ENABLE_TESTS=ON \
  -D Tpetra_ALLOCATE_IN_SHARED_SPACE=ON \
  -D Tpetra_ENABLE_CUDA=ON \
  -D Tpetra_ENABLE_SERIAL=ON \
 -D Trilinos_ENABLE_Sacado=ON \
 -D Sacado_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Amesos2=ON \
 -D Amesos2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Teuchos=ON \
 -D Teuchos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Ifpack2=ON \
 -D Ifpack2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Belos=ON \
 -D Belos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Stokhos=ON \
 -D Stokhos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Phalanx=ON \
 -D Phalanx_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Zoltan2=ON \
 -D Zoltan2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Anasazi=ON \
 -D Trilinos_ENABLE_MueLu=ON \
 -D MueLu_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Panzer=ON \
 -D Panzer_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Intrepid2=ON \
 -D Intrepid2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_SEACAS=OFF \
 -D Trilinos_ENABLE_Zoltan2Sphynx=OFF \
 -D Trilinos_ENABLE_ShyLU_NodeTacho=OFF \
$TRILINOS_DIR
```

**Comment by ndellingwood** [comment]
Just an added note (to reduce sifting through the configuration line) that UVM was enabled with these options set:
```
  -D Kokkos_ENABLE_CUDA_UVM=ON \
  -D KokkosKernels_INST_MEMSPACE_CUDAUVMSPACE=ON \
  -D Tpetra_ALLOCATE_IN_SHARED_SPACE=ON \
```

**Comment by cgcgcg** [comment]
@ndellingwood Could you let me know once this build posts again?

**Comment by ndellingwood** [comment]
@cgcgcg will do, thanks for the fast PR!

**Comment by ndellingwood** [comment]
@cgcgcg your fix #12723 resolved the compilation errors, thanks!
